src/main/cpp/src/hive_hash.cu

```diff
@@ -469,6 +572,42 @@ void check_nested_depth(cudf::table_view const& input)
   }
 }
 
+void flatten_table(std::vector<col_info>& col_infos,
+                   std::vector<cudf::column_view>& basic_cvs,
+                   cudf::table_view const& input,
+                   std::vector<cudf::size_type>& column_map,
+                   rmm::cuda_stream_view const& stream)
+{
+  using column_processer_fn_t = std::function<void(cudf::column_view const&)>;
+
+  // Pre-order traversal
+  column_processer_fn_t flatten_column = [&](cudf::column_view const& col) {
+    auto const type_id = col.type().id();
+    if (type_id == cudf::type_id::LIST) {
+      // Nested size will be updated separately for each row.
+      col_infos.emplace_back(col_info{type_id, -1});
```

Comment:
Add a constructor for col_info, and here `emplace_back` passes the parameters of the constructor.
emplace_back will call the constructor automatically and avoid the calling of the copy constructor.